### DISCUSSION TITLE
Zellij only starts as root

### DISCUSSION THREAD

**Discussion by riley-martin** [zellij-org/zellij#2148] [Q&A]
Starting zellij as a normal user always fails, with fairly cryptic error messages:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }', /home/riley/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-utils-0.34.4/src/consts.rs:91:43
stack backtrace:
   0:     0x559247220883 - <unknown>
   1:     0x559246ed576e - <unknown>
   2:     0x5592471f2f84 - <unknown>
   3:     0x559247221d3f - <unknown>
   4:     0x55924722193f - <unknown>
   5:     0x5592472229d8 - <unknown>
   6:     0x5592472224a2 - <unknown>
   7:     0x55924722240c - <unknown>
   8:     0x5592472223e1 - <unknown>
   9:     0x559246da4e92 - <unknown>
  10:     0x559246da5132 - <unknown>
  11:     0x559246e032ff - <unknown>
  12:     0x5592474f7fbd - <unknown>
  13:     0x55924748814e - <unknown>
  14:     0x55924747beaa - <unknown>
  15:     0x55924747a5d3 - <unknown>
  16:     0x55924747b639 - <unknown>
  17:     0x7f250ec32790 - <unknown>
  18:     0x7f250ec3284a - __libc_start_main
  19:     0x559246e05335 - <unknown>
  20:                0x0 - <unknown>
```
Running it with `strace`, it appears to be trying to create a directory inside `/run/user/1000`, which is not present, and then gets a permission denied error while trying to create `/run/user/1000`. [Here is the strace output](https://gist.github.com/riley-martin/7205368394af622e0a3b4b381a4f7a44).  
Running as root works fine, I guess it has permissions to create the necessary files.  
I am not sure why this directory is not present, and it may not be zellij's problem, but it had been working several days ago.  
I am running Artix Linux with Zellij version 0.34.4.  

EDIT: Just noticed that this problem only occurs when logging in with mosh, not ssh. I am pretty sure it is my problem (maybe PAM isn't set up right?), but if anyone knows more about it I would be glad to hear it.

**Comment by imsnif** [comment]
I think @raphCode has seen this sort of thing before? (or am I mixing things up?)

Anyway, even though the trace mentions `PermissionDenied` I think we could do with a better error message here...

**Reply by riley-martin** [reply]
> Anyway, even though the trace mentions PermissionDenied I think we could do with a better error message here...  

It would be nice to not need `strace` to see what file it is accessing that causes the error :)

**Comment by raphCode** [comment]
I think these may be related:
https://github.com/zellij-org/zellij/issues/879
https://github.com/zellij-org/zellij/issues/2045

https://github.com/zellij-org/zellij/issues/490
https://github.com/zellij-org/zellij/issues/919
